src/components/Trade/hooks/useSwapper/useSwapper.ts

```diff
@@ -116,20 +120,28 @@ export const useSwapper = () => {
     })
 
     if (!minimumQuote) return
+    //use gas estimate from quote if estimatedGasFees var is not populated yet
+    const quoteEstimatedGasFees = fromBaseUnit(minimumQuote?.feeData?.fee || 0, feeAsset.precision)
+
+    // to account for gas fees when trading Max ETH
+    const gasDeduction =
+      sellAsset && feeAsset && feeAsset.caip19 === sellAsset.currency.caip19
+        ? bnOrZero(estimatedGasFees || quoteEstimatedGasFees)
+        : bnOrZero(0)
```

Comment:
I know we do `bnOrZero(0)` a lot around the codebase but that's wrong, it doesn't make sense to fallback to zero if you're already passing zero! What about `bn(0)` ? Also imported from `src/lib/bignumber/bignumber.ts`